api/src/main/java/io/strimzi/api/kafka/model/KafkaConnectorSpec.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2019, Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.api.kafka.model;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+import io.strimzi.crdgenerator.annotations.Description;
+import io.strimzi.crdgenerator.annotations.Minimum;
+import io.sundr.builder.annotations.Buildable;
+import lombok.EqualsAndHashCode;
+
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static java.util.Collections.emptyMap;
+
+@Buildable(
+        editableEnabled = false,
+        generateBuilderPackage = false,
+        builderPackage = "io.fabric8.kubernetes.api.builder"
+)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonPropertyOrder({"connectCluster", "class", "tasksMax", "topics", "config"})
+@EqualsAndHashCode
+public class KafkaConnectorSpec implements Serializable, UnknownPropertyPreserving {
+    private static final long serialVersionUID = 1L;
+
+    private String className;
+    private Integer tasksMax;
+    private List<String> topics;
+    private List<KafkaConnectorConfig> config;
+    private Map<String, Object> additionalProperties;
+
+    @Description("The Class for the Kafka Connector")
+    @JsonProperty("class")
+    public String getClassName() {
+        return className;
+    }
+
+    @Description("The maximum number of tasks for the Kafka Connector")
+    @Minimum(1)
+    public Integer getTasksMax() {
+        return tasksMax;
+    }
```

Comment:
What should be the behaviour if these are also configured via `connector.class` or `tasks.max` in the `config` property? Should these take precedence (perhaps with a warning logged)?

Reply:
It's a good question. The intention was having all the 'mandatory' attributes for a connector in the spec, like the tasks. Maybe the warning it's a good idea @tombentley

Reply:
@tombentley maybe we can handle this with "forbidden" configuration parameters as we do for `Kafka` config for example?